tests/integration/standard/test_custom_protocol_handler.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2013-2015 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest  # noqa
+
+from cassandra.protocol import ProtocolHandler, ResultMessage, UUIDType, read_int, EventMessage
+from cassandra.query import tuple_factory
+from cassandra.cluster import Cluster
+from tests.integration import use_singledc, PROTOCOL_VERSION, execute_until_pass
+from tests.integration.datatype_utils import update_datatypes, PRIMITIVE_DATATYPES, get_sample
+import time
+import uuid
+
+
+def setup_module():
+    use_singledc()
+    update_datatypes()
+
+
+class CustomProtocolHandlerTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.cluster = Cluster(protocol_version=PROTOCOL_VERSION)
+        cls.session = cls.cluster.connect()
+        cls.session.execute("CREATE KEYSPACE custserdes WITH replication = { 'class' : 'SimpleStrategy', 'replication_factor': '1'}")
+        cls.session.set_keyspace("custserdes")
+
+    @classmethod
+    def tearDownClass(cls):
+        execute_until_pass(cls.session, "DROP KEYSPACE custserdes")
+        cls.cluster.shutdown()
+
+    def test_custom_raw_uuid_row_results(self):
+        """
+        Test to validate that custom protocol handlers work with raw row results
+
+        Connect and validate that the normal protocol handler is used.
+        Re-Connect and validate that the custom protocol handler is used.
```

Comment:
I think with the API changes we do not need to reconnect Clusters/Sessions. We can just replace the ``session.client_protocol_handler`` for each phase.